File: LeetCodePython/py67.py
# 二进制求和
class Solution:
    def addBinary(self, a: str, b: str) -> str:
        result = ""
        i = len(a) - 1
        j = len(b) - 1
        carry = 0
        while (i>=0 or j>=0):
            temp = carry
            temp += int(a[i]) if i>=0 else 0
            temp += int(b[j]) if j>=0 else 0
            result = str(temp%2) + result
            carry = temp // 2
            i -= 1
            j -= 1
        
        if carry:
            result = '1' + result
        
        return result

File: LeetCodeCPlusPlus/py67.cpp
// 二进制求和
#include <string>
using namespace std;
class Solution {
public:
    string addBinary(string a, string b) {
        string s;
		s.reserve(a.size() + b.size());
		int c = 0, i = a.size() - 1, j = b.size() - 1;
		while(i >= 0 || j >= 0 || c == 1)
		{
			c += i >= 0 ? a[i--] - '0' : 0;
			c += j >= 0 ? b[j--] - '0' : 0;
			s.push_back((c & 1) + '0');
			c >>= 1;
		}
		reverse(s.begin(), s.end());
		return s;
    }
};